src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioWorkspaceImpl.cs

```diff
@@ -644,6 +644,90 @@ protected override void ApplyAnalyzerReferenceRemoved(ProjectId projectId, Analy
             }
         }
 
+        protected override void ApplyProjectChanges(ProjectChanges projectChanges)
+        {
+            base.ApplyProjectChanges(projectChanges);
+
+            // After normal project changes have been applied, we need to apply changes for
+            // generated razor documents that are mapped to a location in a file outside of our workspace.
+            // Since these generated documents have CanApplyChange() == false, they will not
+            // be applied during the normal ApplyProjectChanges workflow.
```

Comment:
Why are they returning CanApplyChange == false now? Should that be true instead? I'm surprised this code ever got this far as opposed to crashing earlier. :smile:

Reply:
they always returned canapplychange == false - the underlying generated .g.cs files were always unchangeable (and still are).  However we can modify the .razor file.

Reply:
I guess that does mean that rename is not checking if the documents being rename are changeable